frontend/src/views/DataView.svelte

```diff
@@ -0,0 +1,118 @@
+<style>
+  .content {
+    position: sticky;
+    top: 0;
+    height: 100%;
+  }
+  .breadcrumb {
+    padding-bottom: 1em;
+    background: var(--main-bg-color);
+  }
+  button {
+    margin-bottom: 0;
+    border: 1px solid white;
+  }
+  .content hr {
+    display: block;
+    height: 1px;
+    border: 0;
+    border-top: 1px solid white;
+    margin-top: -1px;
+    padding: 0;
+  }
+</style>
+
+<script>
+  import { selectedResource } from "../stores";
+  import Breadcrumb from "../utils/Breadcrumb.svelte";
+  import AssemblyView from "./AssemblyView.svelte";
+  import DecompilationView from "./DecompilationView.svelte";
+  import HexView from "./HexView.svelte";
+  import TextView from "./TextView.svelte";
+  import SearchBar from "../utils/SearchBar.svelte";
+
+  export let dataLenPromise, scrollY, resourceNodeDataMap, resources;
+  let display_type = "hex";
+  let hasTextView = false;
+  let hasAsmView = false;
+  let hasDecompView = false;
+  let dataSearchResults = {};
+  let searchFunction;
+
+  function checkTags() {
+    display_type = "hex";
+    hasTextView = ["ofrak.core.binary.GenericText"].some((tag) =>
+      $selectedResource.has_tag(tag)
+    );
+    hasAsmView = [
```

Comment:
I think it can be made simpler to add more tab-able views, if all you need to do is add an entry to a table for the new view and what tags it supports (and also implement the view itself, but that's always the case). This is only relevant if you think it's likely that there are more views of this nature we might realistically add soon-ish. The only one I can think of at the moment is a CFG view for `ComplexBlock`s, and I'm not sure how realistic that is. However if, after implementing this tabbing feature, you are bursting with ideas for how we can add to it, then I encourage implementing it with scalability in mind. If not though, then the current implementation is perfectly fine and even if it were totally inconvenient for adding new tabs, it can be changed later, so you can ignore the rest of this comment.

The idea for how to implement this is to replace this logic (and the svelte if/else chain later) with a mapping from each of the possible content views to the group of tags they are applicable for. Then the list of applicable content views is built from checking each of them against the actual resource's tags, the group of "tab" buttons is built from that list with a Svelte `{#each ...}`, and each button's on-click assigns the corresponding content view Svelte component to a common variable, which can be used by Svelte to show the selected view without an if/else chain with:

```
<svelte:component
              this="{selectedContentView}"
              bind:searchFunction="{searchFunction}"
            />
```

(similar to how we use a variable to hold the type of the view in the bottom left pane, and change that variable, rather than switch on the different cases of what it might be)

Reply:
good idea, implemented.